File: solutions/0020-valid-parentheses/solution.cpp
class Solution {
public:
    bool isValid(string st) {
        stack<int> s;
        for(char a:st)
        {
            if(a=='(' || a=='[' || a=='{')
                s.push(a);
            else
            {
                if(s.empty()) return false;
                if(a=='}' && '{'!=s.top())
                    return false;
                if(a==']' && '['!=s.top())
                    return false;
                if(a==')' && '('!=s.top())
                    return false;
                s.pop();
            }
        }
        return s.empty();
    }
};

File: solutions/0020-valid-parentheses/solution.java
class Solution {
    public boolean isValid(String st) {
        Stack<Character>s=new Stack<>();
        for(char a:st.toCharArray())
        {
            if(a=='{' || a=='[' || a=='(')
                s.push(a);
            else
            {
                if(s.isEmpty()) return false;
                if(a=='}' && s.pop()!='{') return false;
                if(a==']' && s.pop()!='[') return false;
                if(a==')' && s.pop()!='(') return false;
            }
        }
        return s.isEmpty();
    }
}
